templates/vue/src/components/modals/UserSettingsModal/AvatarForm.vue

```diff
@@ -0,0 +1,453 @@
+        { value: "ShortHairShortFlat", text: "ShortHairShortFlat" },
+        { value: "ShortHairShortRound", text: "ShortHairShortRound" },
+        { value: "ShortHairShortWaved", text: "ShortHairShortWaved" },
+        { value: "ShortHairSides", text: "ShortHairSides" },
+        { value: "ShortHairTheCaesar", text: "ShortHairTheCaesar" },
+        { value: "ShortHairTheCaesarSidePart", text: "ShortHairTheCaesarSidePart" },
+      ],
+      topColorOptions: [
+        { value: "Black", text: "Black" },
+        { value: "Blue01", text: "Blue01" },
+        { value: "Blue02", text: "Blue02" },
+        { value: "Blue03", text: "Blue03" },
+        { value: "Gray01", text: "Gray01" },
+        { value: "Gray02", text: "Gray02" },
+        { value: "Heather", text: "Heather" },
+        { value: "PastelBlue", text: "PastelBlue" },
+        { value: "PastelGreen", text: "PastelGreen" },
+        { value: "PastelOrange", text: "PastelOrange" },
+        { value: "PastelRed", text: "PastelRed" },
+        { value: "PastelYellow", text: "PastelYellow" },
+        { value: "Pink", text: "Pink" },
+        { value: "Red", text: "Red" },
+        { value: "White", text: "White" },
+      ],
+    }
+  },
+  computed: {
+    ...mapState(["avatar"]),
+  },
+  mounted() {
+    this.setExistingAvatar()
+  },
+  methods: {
+    ...mapActions(["updateAvatar"]),
+    setExistingAvatar() {
+      if (!this.avatar || Object.keys(this.avatar).length === 0) return
+      this.isCircle = this.avatar.isCircle
+      this.circleColor = this.avatar.circleColor
+      this.accessoriesType = this.avatar.accessoriesType
+      this.clotheType = this.avatar.clotheType
+      this.clotheColor = this.avatar.clotheColor
+      this.eyebrowType = this.avatar.eyebrowType
+      this.eyeType = this.avatar.eyeType
+      this.facialHairColor = this.avatar.facialHairColor
+      this.facialHairType = this.avatar.facialHairType
+      this.graphicType = this.avatar.graphicType
+      this.hairColor = this.avatar.hairColor
+      this.mouthType = this.avatar.mouthType
+      this.skinColor = this.avatar.skinColor
+      this.topType = this.avatar.topType
+      this.topColor = this.avatar.topColor
+    },
+    saveAvatar() {
+      const newAvatar = {}
+      newAvatar.isCircle = this.isCircle
+      newAvatar.circleColor = this.circleColor
+      newAvatar.accessoriesType = this.accessoriesType
+      newAvatar.clotheType = this.clotheType
+      newAvatar.clotheColor = this.clotheColor
+      newAvatar.eyebrowType = this.eyebrowType
+      newAvatar.eyeType = this.eyeType
+      newAvatar.facialHairColor = this.facialHairColor
+      newAvatar.facialHairType = this.facialHairType
+      newAvatar.graphicType = this.graphicType
+      newAvatar.hairColor = this.hairColor
+      newAvatar.mouthType = this.mouthType
+      newAvatar.skinColor = this.skinColor
+      newAvatar.topType = this.topType
+      newAvatar.topColor = this.topColor

```

Comment:
Allocating all these props inside the {} would make the code look cleaner as you will not have to access newAvatar all the time.

Reply:
Again, I suggest saving the whole thing rather than line by line. Maybe you can just create a `preferences` prop that saves all of these into an object so you can just pass that in rather than set it line by line.